### PR TITLE
add double caching for yelp polarity to speed up extracted reading.

### PR DESCRIPTION
**Pull request by erip**
Reference #1526

### CODE REVIEW

File: torchtext/datasets/yelpreviewpolarity.py [hunk 1]
```diff
@@ -33,22 +33,28 @@
...
+    cache_compressed_dp = url_dp.on_disk_cache(
         filepath_fn=lambda x: os.path.join(root, _PATH),
-        hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
+        hash_dict={os.path.join(root, _PATH): MD5},
+        hash_type="md5"
     )
-    cache_dp = GDriveReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
-    cache_dp = FileOpener(cache_dp, mode="b")
+    cache_compressed_dp = GDriveReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
+    cache_compressed_dp = FileOpener(cache_compressed_dp, mode="b")
```
Comment (parmeet) [inline torchtext/datasets/yelpreviewpolarity.py:49]: For consistency, can we move FileOpener after creating on_disk_cache datapipe like done for AmazonReviewPolarity (https://github.com/pytorch/text/blob/0f7f859e412fba4a31852c1a84801a182e636fde/torchtext/datasets/amazonreviewpolarity.py#L49)

**Review by parmeet — COMMENTED** [review]
Overall LGTM! For testing, could you please do so by removing the local cache and doing it from scratch, I have seen tests passing in weird ways by the presence of cache in CI and locally :). Thanks!

File: torchtext/datasets/yelpreviewpolarity.py (hunk 1 above)
Comment (erip) [reply torchtext/datasets/yelpreviewpolarity.py:49]: Done!

Comment (Nayef211) [reply torchtext/datasets/yelpreviewpolarity.py:49]: Just wondering, is there any benefit to calling the `FileOpener` datapipe before calling `end_caching`? Or is this purely for consistency?

**Review by parmeet — APPROVED** [review]
LGTM!

**Pull request merged**